### PR TITLE
Task/rhornung67/fix readme

### PR DESCRIPTION
**Pull request by rhornung67**
Mostly cosmetic changes:
- Fixed some broken links, etc. in README file
- Updated copyright year in all files, added missing copyright info where needed
- Enforced consistent placement of doxygen file comment, copyright info, include guard for header files
- Changed some header include guards that still used 'HXX'
- Fixed a few places where config header was included after a macro conditional for constant defined in config header
- Removed redundant header inclusions in a few places
- Etc.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -19,12 +19,14 @@ Contributors include:
```
Comment (rhornung67) [inline LICENSE]: Note: All contributors need to be included in LICENSE file per LLNL release policy.

File: README.md [hunk 2]
```diff
@@ -110,31 +110,22 @@ The original developers of RAJA are:
...
-
-  * David Beckingsale (beckingsale1@llnl.gov)
-  * Jason Burmark (burmark1@llnl.gov)
-  * Holger Jones (jones19@llnl.gov)
-  * Will Killian (killian4@llnl.gov)
-  * Adam Kunen (kunen1@llnl.gov)
-  * Olga Pearce (pearce8@llnl.gov)
-  * David Poliakoff (poliakoff1@llnl.gov)
-  * Tom Scogland (scogland1@llnl.gov)
+Please see the {RAJA Contributors Page](https://github.com/LLNL/RAJA/graphs/contributors), to see the full list of contributors to the project.
```
Comment (rhornung67) [inline README.md]: Rather than maintain contributor list in multiple places, I just added a link to it here.

**Comment by trws** [comment]
I don't see a problem with any of this, but it occurs to me to wonder whether we can use a by-reference copyright/license clause on the files rather than pasting the whole thing into each one? It's a lot of extra places to update, a lot of extra text, and a reference to the main license file should be sufficient unless policy says "thou shalt" or similar.

**Comment by davidbeckingsale** [comment]
By reference would be really nice, I don't see the benefit in having it in full in every file.

**Comment by rhornung67** [comment]
@trws @davidbeckingsale I agree that including copyright legalese in every file is onerous and should be avoided if possible. I recall discussing this with someone in the IPO office a couple of years ago and was under the impression that it was necessary. The pattern is also followed by other LLNL projects, such as BLT https://github.com/LLNL/blt and Conduit https://github.com/LLNL/conduit. 

At any rate, I'll look into this for clarification of what is actually required.

**Comment by trws** [comment]
It may be license dependent, I recall that Mozilla added specific 
support for it in MPLv2, but I don’t know the lab specific 
requirements or if special handling is necessary for BSD.

File: LICENSE [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 RAJA: ................................, version 0.3.1
 
-Copyright (c) 2016, Lawrence Livermore National Security, LLC. 
+Copyright (c) 2016-17, Lawrence Livermore National Security, LLC.
```
Comment (davidbeckingsale) [inline LICENSE]: Can't the year just be 2017?

**Comment by rhornung67** [comment]
@trws @davidbeckingsale I'm going to hold off on merging this until I learn the requirements for including copyright information in source code files.

**Comment by rhornung67** [comment]
@trws and @davidbeckingsale at present, there is no clear policy about inclusion of copyright in files. I've reduced what was in each file to what I've verified to be the minimum required per IPO.  So, we'll go with this. Please review, and if OK, I'll merge when CI passes.

**Pull request merged**